### PR TITLE
add naivest entry API to VecMap

### PR DESCRIPTION
**Pull request by Gankra**
Building over night, posting for review now. Presumably not much should need change.

I consider this necessary to move forward with a proper stabilization of the API.

r? @huonw

### CODE REVIEW

File: src/libcollections/vec_map.rs [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 #![allow(missing_docs)]
 
+pub use self::Entry::*;
```
Comment (Gankro) [inline src/libcollections/vec_map.rs]: Hmm, I basically was copy-pasting from BTreeMap, I didn't even notice the `pub`.

I'm going to refactor this all in my next PR anyway.

Comment (frewsxcv) [inline src/libcollections/vec_map.rs]: Why not get rid of it now? One fewer thing to forget

Comment (Gankro) [inline src/libcollections/vec_map.rs]: Won't be home to fix this for a while.

**Comment by alexcrichton** [comment]
(sorry about the weird activity, try to kick homu)

**Pull request merged**